### ISSUE TITLE
Build failures relating to shm_open and shm_unlink

### ISSUE THREAD

**Issue by pprindeville** [openwrt/libubox#9]
Looking at these builds I'm seeing:

```
====== Make errors from logs/package/libs/libubox/host-compile.txt ======
[41/46] Building C object CMakeFiles/blobmsg_json-static.dir/blobmsg_json.c.o
[42/46] Linking C static library libblobmsg_json.a
[43/46] Building C object CMakeFiles/jshn.dir/jshn.c.o
[44/46] Linking C executable jshn
FAILED: jshn 
: && /__w/openwrt/openwrt/openwrt/staging_dir/host/bin/gcc -O2 -I/__w/openwrt/openwrt/openwrt/staging_dir/host/include -I/__w/openwrt/openwrt/openwrt/staging_dir/hostpkg/include -I/__w/openwrt/openwrt/openwrt/staging_dir/target-mips-openwrt-linux-musl_musl/host/include -DNDEBUG -L/__w/openwrt/openwrt/openwrt/staging_dir/host/lib -L/__w/openwrt/openwrt/openwrt/staging_dir/hostpkg/lib -L/__w/openwrt/openwrt/openwrt/staging_dir/target-mips-openwrt-linux-musl_musl/host/lib   -rdynamic CMakeFiles/jshn.dir/jshn.c.o -o jshn  -Wl,-rpath,/__w/openwrt/openwrt/openwrt/build_dir/hostpkg/libubox-2023-11-27-d4c3066e:  libblobmsg_json.so  -Wl,-Bstatic  -ljson-c  -Wl,-Bdynamic  libubox.so && :
/usr/bin/ld: libubox.so: undefined reference to `shm_unlink'
/usr/bin/ld: libubox.so: undefined reference to `shm_open'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
make[3]: *** [Makefile:114: /__w/openwrt/openwrt/openwrt/build_dir/hostpkg/libubox-2023-11-27-d4c3066e/.built] Error 1
time: package/libs/libubox/host-compile#5.00#1.95#9.01
```

which implies that `-lrt` should be handled as:

```
TARGET_LINK_LIBRARIES(ubox rt)
```

unconditionally.

https://github.com/openwrt/openwrt/actions/runs/7009125615/job/19067004747?pr=13780
https://github.com/openwrt/openwrt/actions/runs/7009125615/job/19067008357?pr=13780

**Comment by tateg** [comment]
I started seeing the exact same build failures starting at commit: `8a5a4319a85cdfe54998c8f15bb29cd81b2b2c01`

**Comment by pprindeville** [comment]
Fixed by commit e80dc00.
